src/release/releaseHelper.d

```diff
@@ -0,0 +1,234 @@
+
+        string prev_relnotes;
+
+        if (this.tag_prev.length > 0)
+            prev_relnotes = getTagMessage(this.tag_prev);
+
+        string tag_msg;
+
+        // Patch versions don't get release notes (yet)
+        if (tag_version.patch > 0)
+            tag_msg = tag_version.toString;
+        else
+            tag_msg = buildReleaseNotes(this.tag_prev, prev_relnotes);
+
+        assert(tag_msg.length > 0, "No release notes found?!");
+
+        auto proc = pipeShell(this.command, Redirect.all);
+
+        proc.stdin.write(tag_msg);
+        proc.stdin.flush();
+        proc.stdin.close();
+
+        string ret, line;
+
+        wait(proc.pid);
+
+        while ((line = proc.stdout.readln()) !is null)
+            ret ~= line;
+
+        return ret;
+    }
+
+    /// Returns description of this action
+    override string description ( ) const
+    {
+        import std.format;
+        return format("Creating annotated tag %s", this.tag_version);
+    }
+
+    /// Returns command of this action
+    override string command ( ) const
+    {
+        import std.format;
+        return format("git tag -F- %s %s",
+                      this.tag_version,
+                      this.tag_reference);
+    }
+}
+
+/*******************************************************************************
+
+    Range filter function to only match files ending with the given string
+
+    Params:
+        range = range to filter
+        match_str = string to filter for
+
+    Returns:
+        filtered range
+
+*******************************************************************************/
+
+private auto match ( R ) ( R range, string match_str )
+{
+    import std.algorithm;
+
+    return range.filter!(a=>a.endsWith(match_str));
+}
+
+/*******************************************************************************
+
+    Creates release notes, including inherited notes.
+
+    Params:
+        previous_version = the previous release which this one inherited from
+        previous_notes = the previous releases' notes
+
+    Returns:
+        this release' notes
+
+*******************************************************************************/
+
+string buildReleaseNotes ( string previous_version, string previous_notes  )
+{
+    import release.gitHelper;
+
+    import std.file;
+    import std.algorithm;
+    import std.format;
+    import std.range;
+    import std.array : join;
+
+    import std.stdio;
+
+    immutable MigrationHeader = "Migration Instructions\n----------------------\n\n";
+    immutable DeprecationsHeader = "Deprecations\n------------\n\n";
+    immutable FeaturesHeader = "Features\n--------\n\n";
+
+    auto files = dirEntries("relnotes/", SpanMode.shallow)
+                        .map!(a=>a.name)
+                        .filter!(a=>previous_version.length == 0 ||
+                                    !isAncestor(getLastCommitOf(a),
+                                                previous_version))
```

Comment:
Not going to ask to change it right now because it slipped through my attention in earlier PRs but worth noting that this is not indentation scheme we normally use - it should always be single ident per line break / semantical nesting.